### PR TITLE
Support for no storage account scenario. (#5922)

### LINKED ISSUE #5922: Add support for AzureWebJobsSecretStorageType=none
For customers not using AzureWebjobsStorage / AzureWebJobsSecretStorageSas there is no where to store function/host keys. This causes failures anytime an attempt is made to access secrets, e.g., during synctriggers, secrets management admin APIs, etc.

Proposal is to support AzureWebJobsSecretStorageType=none, For apps using this configuration we will not generate or persist any keys.

Instead all authentication will be based on jwt tokens based on AzureWebEncryptionKey.

@mathewc / @balag0 for additional details.

### PR DESCRIPTION
**Pull request by kashimiz**
### Issue describing the changes in this PR

resolves #5922

### Pull request checklist

* [ ] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
    * [X] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
    * [X] Otherwise: Backport tracked by issue/PR #8396 
* [X] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Security/Authentication/Keys/AuthenticationLevelHandler.cs [hunk 1]
```diff
@@ -84,22 +83,25 @@ protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
 
         internal static Task<(string, AuthorizationLevel)> GetAuthorizationKeyInfoAsync(HttpRequest request, ISecretManagerProvider secretManagerProvider)
         {
-            // first see if a key value is specified via headers or query string (header takes precedence)
-            string keyValue = null;
-            if (request.Headers.TryGetValue(FunctionsKeyHeaderName, out StringValues values))
+            if (secretManagerProvider.SecretsEnabled)
```
Comment (liliankasem) [inline src/WebJobs.Script.WebHost/Security/Authentication/Keys/AuthenticationLevelHandler.cs:86]: Can we reverse the if condition and return at the top if secrets is not enabled?

File: release_notes.md [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (liliankasem) [inline release_notes.md]: nit: extra space?

**Review by liliankasem — APPROVED** [review]
lgtm, please make sure to test with a private site extension too!

**Review by soninaren — APPROVED** [review]
Nit change pointed by lilian, looks good otherwise

File: test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SWAEndToEndTests.cs [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        }
+
+        [Fact]
+        public async Task InvokeFunction_AnonymousLevel_Succeeds()
+        {
+            var code = "test";
+
+            // send along a code query param, and ensure that the invocation succeeds
+            // the token isn't interpreted as a function key in this case since keys are disabled
+            var content = new StringContent(JsonConvert.SerializeObject(new { scenario = "swa" }));
```
Comment (madelinegordon) [inline test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SWAEndToEndTests.cs]: nit: same as above about more descriptive string for "swa"

File: test/WebJobs.Script.Tests.Integration/TestScripts/CSharp/HttpTrigger-Scenarios/run.csx [hunk 4]
```diff
@@ -13,6 +14,10 @@ public static async Task<HttpResponseMessage> Run(HttpRequestMessage req, Execut
 
     switch (scenario)
     {
+        case "swa":
```
Comment (madelinegordon) [inline test/WebJobs.Script.Tests.Integration/TestScripts/CSharp/HttpTrigger-Scenarios/run.csx]: nit: could change to "staticWebApps" or add comment for easier readability?

**Review by madelinegordon — APPROVED** [review]

File: release_notes.md (hunk 2 above)
Comment (kashimiz) [reply release_notes.md]: Huh, not sure how that got in there. Thanks for the catch!

File: test/WebJobs.Script.Tests.Integration/TestScripts/CSharp/HttpTrigger-Scenarios/run.csx (hunk 4 above)
Comment (kashimiz) [reply test/WebJobs.Script.Tests.Integration/TestScripts/CSharp/HttpTrigger-Scenarios/run.csx]: Now that you bring this up, I remember it took me several exposures to "SWA" to remember it stood for "static web apps." Good call on avoiding confusion.

File: test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SWAEndToEndTests.cs (hunk 3 above)
Comment (kashimiz) [reply test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SWAEndToEndTests.cs]: Same as below; changing to `staticWebApp`.

File: src/WebJobs.Script.WebHost/Security/Authentication/Keys/AuthenticationLevelHandler.cs (hunk 1 above)
Comment (kashimiz) [reply src/WebJobs.Script.WebHost/Security/Authentication/Keys/AuthenticationLevelHandler.cs:86]: Spoke with Lilian about this offline; reversing the condition would double the return statement, so keeping as-is.

**Pull request merged**